llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp

```diff
@@ -2570,8 +2569,7 @@ bool isArgPassedInSGPR(const CallBase *CB, unsigned ArgNo) {
     return CB->paramHasAttr(ArgNo, Attribute::InReg) ||
            CB->paramHasAttr(ArgNo, Attribute::ByVal);
   default:
-    // TODO: Should calls support inreg for SGPR inputs?
-    return false;
+    return CB->paramHasAttr(ArgNo, Attribute::InReg);
```

Comment:
Don't think the outgoing argument case is tested here

Reply:
Did you mean adding testcases for pointer arguments, such as:
int foo (int *A) {
  *A = 1;
}

Reply:
No, I mean for calls. e.g. `call @foo(i32 inreg %outgoing0, <2 x double> inreg %outgoing1)`

Reply:
Added a testcase.